libraries/DataFlash/DataFlash.cpp

```diff
@@ -351,6 +351,11 @@ bool DataFlash_Class::fill_log_write_logstructure(struct LogStructure &logstruct
             break;
         }
     }
+
+    if (!f) {
```

Comment:
it won't. And this is common idiom to check if the for loop actually found something (or that log_write_fmts was null to begin with).